test/norm_num.lean

```diff
@@ -219,6 +219,31 @@ section norm_num_cmd_variable
 
 end norm_num_cmd_variable
 
+section norm_num_erase
+
+example : 3 ^ 3 + 4 = 31 := by norm_num1; with_reducible rfl
+
+/- should error:
+type mismatch
+  HEq.rfl
+has type
+  HEq ?m.38198 ?m.38198 : Prop
+but is expected to have type
+  3 ^ 3 + 4 = 31 : Prop
```

Comment:
@thorimur Can you please use `successIfFail` here, even though there's no good way to make the `with_msg` part work?

Reply:
Hmm, I'm having a bit of trouble translating the monad operation `successIfFail` into this context. Would a syntactic `guard_target` be alright? Or is admit frowned upon?
```
attribute [-norm_num] Mathlib.Meta.NormNum.evalPow in
example : 3 ^ 3 + 4 = 31 := by
  norm_num1
  guard_target =ₛ 3 ^ 3 + 4 = 31
  admit
```

Reply:
I think that's a good approach!  (Also you can probably change the `admit` to `rfl`, if you want to get rid of the warning.)